src/pages/components/OverlayListItem.qml

```diff
@@ -0,0 +1,56 @@
+/****************************************************************************
+**
+** Copyright (C) 2020 Open Mobile Platform LLC
+**
+****************************************************************************/
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.6
+import Sailfish.Silica 1.0
+
+BackgroundItem {
+    id: root
+
+    property alias text: label.text
+    property alias iconSource: icon.source
+    property int horizontalOffset
+    property int iconWidth
+
+    width: parent.width
+
+    Item {
+        id: iconContainer
+
+        width: iconWidth
+        height: parent.height
+
+        Icon {
+            id: icon
+
+            anchors {
+                centerIn: parent
+                horizontalCenterOffset: horizontalOffset
+            }
+            highlighted: root.highlighted
+            color: Theme.primaryColor
+        }
```

Comment:
As label is anchored to the right of the iconContainer (reserving the space for the icon) it is cleaner to close icoContainer  here.